src/main/scala/de/upb/cs/swt/delphi/crawler/io/swagger/client/model/Instance.scala

```diff
@@ -0,0 +1,52 @@
+/**
+ * NOTE: This class is auto generated by the akka-scala (beta) swagger code generator program.
+ * https://github.com/swagger-api/swagger-codegen
+ * For any issue or feedback, please open a ticket via https://github.com/swagger-api/swagger-codegen/issues/new
+ */
+
+package de.upb.cs.swt.delphi.crawler.io.swagger.client.model
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import spray.json._
+
+trait JsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
+  implicit val componentTypeFormat = new JsonFormat[InstanceEnums.ComponentType] {
+    def write(compType : InstanceEnums.ComponentType) = JsString(compType.toString)
+
+    def read(value: JsValue) = value match {
+      case JsString(s) => s match {
+        case "Crawler" => InstanceEnums.ComponentType.Crawler
+        case "WebApi" => InstanceEnums.ComponentType.WebApi
+        case "WebApp" => InstanceEnums.ComponentType.WebApp
+        case "DelphiManagement" => InstanceEnums.ComponentType.DelphiManagement
+        case "ElasticSearch" => InstanceEnums.ComponentType.ElasticSearch
+        case x => throw new RuntimeException(s"Unexpected string value $x for component type.")
+      }
+      case y => throw new RuntimeException(s"Unexpected type $y while deserializing component type.")
+    }
+  }
+  implicit val instanceFormat = jsonFormat5(Instance)
+}
+
+final case class Instance (
```

Comment:
Why are all fields of this case class `Option`s?

Reply:
This is the model class that was generated by swagger. I'm not sure why it was generated using all Options, should we modify it manually to not use Options anymore ?

Reply:
The Instance class has now been refactored in all projects projects that depend on the Instance Registry. The only `Option` that still exists is the `ID` attribute, as it needs to be empty for registering.